zone/bonuses.cpp

```diff
@@ -5631,3 +5641,127 @@ void Mob::NegateSpellEffectBonuses(uint16 spell_id)
 		}
 	}
 }
+
+void Mob::CalcHeroicBonuses(StatBonuses* newbon)
+{
+	if (GetHeroicSTR()) {
+		SetHeroicStrBonuses(newbon);
+	}
+	if (GetHeroicSTA()) {
+		SetHeroicStaBonuses(newbon);
+	}
+	if (GetHeroicAGI()) {
+		SetHeroicAgiBonuses(newbon);
+	}
+	if (GetHeroicDEX()) {
+		SetHeroicDexBonuses(newbon);
+	}
+	if (GetHeroicINT()) {
+		SetHeroicIntBonuses(newbon);
+	}
+	if (GetHeroicWIS()) {
+		SetHeroicWisBonuses(newbon);
+	}
+}
+
+void Mob::SetHeroicWisBonuses(StatBonuses* newbon) {
+
+	newbon->heroic_max_mana += (GetCasterClass() == 'W') ? GetHeroicWIS() * RuleR(Character, HeroicWisdomMultiplier) * 10 : 0;
+	newbon->heroic_mana_regen += (GetCasterClass() == 'W') ? GetHeroicWIS() * RuleR(Character, HeroicWisdomMultiplier) / 25 : 0;
+	newbon->HealAmt += GetHeroicWIS() * RuleR(Character, HeroicWisdomIncreaseHealAmtMultiplier);
+
+	if (RuleB(Character, HeroicStatsUseDataBucketsToScale)) {
+		newbon->heroic_max_mana += (GetCasterClass() == 'W') ? GetHeroicWIS() * CheckHeroicBonusesDataBuckets("HWIS-MaxMana") * 10 : 0;
+		newbon->heroic_mana_regen += (GetCasterClass() == 'W') ? GetHeroicWIS() * CheckHeroicBonusesDataBuckets("HWIS-ManaRegen") / 25 : 0;
+		newbon->HealAmt += GetHeroicWIS() * CheckHeroicBonusesDataBuckets("HWIS-HealAmt");
+	}
+}
+
+void Mob::SetHeroicIntBonuses(StatBonuses* newbon) {
+
+	newbon->heroic_max_mana += (GetCasterClass() == 'I') ? GetHeroicINT() * RuleR(Character, HeroicIntelligenceMultiplier) * 10 : 0;
+	newbon->heroic_mana_regen += (GetCasterClass() == 'I') ? GetHeroicINT() * RuleR(Character, HeroicIntelligenceMultiplier) / 25 : 0;
+	newbon->SpellDmg += GetHeroicINT() * RuleR(Character, HeroicIntelligenceIncreaseSpellDmgMultiplier);
```

Comment:
Where's the math coming from

Reply:
Math was moved into these Setters from the original locations client_mods.cpp and bot.cpp, such as:

```	item_bonus += heroic_bonus / 25;``` where we calculate heroic mana regen.